test_tf.py

```diff
@@ -0,0 +1,153 @@
+import os
+import subprocess
+import unittest
+import pytest
+from parameterized import parameterized
+from main import get_policy_check_type, get_required_inputs, get_optional_inputs, get_sub_command
+from main import get_treat_findings_as_non_blocking_flag, build_command, execute_command, set_output
+
+from main import CLI_POLICY_VALIDATOR, POLICY_CHECK_TYPE, VALIDATE_POLICY, CHECK_NO_NEW_ACCESS, CHECK_ACCESS_NOT_GRANTED
+from main import COMMON_REQUIRED_INPUTS, TREAT_FINDINGS_AS_NON_BLOCKING
+
+from unittest.mock import patch
+
+class TfpvTest(unittest.TestCase):
+    # case 1: test_get_type_ENVIRON_NOT_SET: failure expected because required os.environ[] are not set
+    @unittest.expectedFailure
+    def test_get_type_ENVIRON_NOT_SET(self):
+        assertRaises(ValueError, get_policy_check_type)
+
+    # case 2: test_get_type_INVALID_POLICY_CHECK: failure expected because os.environ[] is set to an invalid value
+    @unittest.expectedFailure
+    def test_get_type_INVALID_POLICY_CHECK(self):
+        os.environ[POLICY_CHECK_TYPE] = INVALID_POLICY_CHECK
+        assertRaises(ValueError, get_policy_check_type)
+
+    # case 3, 4, 5: test_get_type_WITH_VALIDATE_POLICY: success with valid POLICY_CHECK_TYPE
+    @parameterized.expand([VALIDATE_POLICY, CHECK_NO_NEW_ACCESS, CHECK_ACCESS_NOT_GRANTED])
+    def test_get_type_WITH_VALID_POLICY_CHECK(self, policy_check_type):
+        os.environ[POLICY_CHECK_TYPE] = policy_check_type
+        policy_type = get_policy_check_type()
+        self.assertEqual(policy_type, policy_check_type)
+
+    # case 6: test_get_required_input_INVALID_POLICY_CHECK: failure expected because an invalid policy_check_type is provided
+    @unittest.expectedFailure
+    def test_get_required_input_INVALID_POLICY_CHECK(self):
+        policy_check = "INVALID_POLICY_CHECK"
+        assert get_required_inputs(policy_check) is None
+
+    # case 7, 8, 9: test_get_required_input_WITH_VALIDATE_POLICY: success as a valid POLICY_CHECK_TYPE is provided: VALIDATE_POLICY
+    @parameterized.expand([VALIDATE_POLICY, CHECK_NO_NEW_ACCESS, CHECK_ACCESS_NOT_GRANTED])
+    def test_get_required_input_WITH_VALID_POLICY_CHECK(self, policy_check_type):
+        result = get_required_inputs(policy_check_type)
+        if policy_check_type == VALIDATE_POLICY:
+            self.assertEqual(result, {"INPUT_TEMPLATE-PATH",  "INPUT_REGION"})
```

Comment:
with parametrized tests, we should be able to pass in expected result as parameters as well, instead of maintaining this if-else branch